deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/LayerConfiguration.java

```diff
@@ -0,0 +1,313 @@
+     *
+     * @param kerasLayerClass  String containing the Keras layer class type
+     * @param kerasConfig      Map containing Keras layer properties
+     * @param isOutput         Whether this is an output layer
+     * @return                 DL4J Layer configuration
+     * @see Layer
+     */
+    public static Layer buildLayer(String kerasLayerClass, Map<String,Object> kerasConfig, boolean isOutput) {
+        Layer layer = null;
+        switch (kerasLayerClass) {
+            case "Dense":
+            case "TimeDistributedDense":
+                layer = buildDenseLayer(kerasConfig);
+                break;
+            case "LSTM":
+                layer = buildGravesLstmLayer(kerasConfig);
+                break;
+            case "Convolution2D":
+                layer = buildConvolutionLayer(kerasConfig);
+                break;
+            case "MaxPooling2D":
+                layer = buildSubsamplingLayer(kerasConfig);
+                break;
+            case "Flatten":
+                log.warn("DL4J adds reshaping layers during model compilation");
+                break;
+            default:
+                throw new IncompatibleKerasConfigurationException("Unsupported keras layer type " + kerasLayerClass);
+        }
+        return layer;
+    }
+
+    /**
+     * Map Keras to DL4J activation functions.
+     *
+     * @param kerasActivation   String containing Keras activation function name
+     * @return                  String containing DL4J activation function name
+     */
+    public static String mapActivation(String kerasActivation) {
+        if (kerasActivation.equals("linear"))
+            return "identity";
+        return kerasActivation;
+    }
+
+    /**
+     * Map Keras to DL4J weight initialization functions.
+     *
+     * @param kerasInit     String containing Keras initialization function name
+     * @return              DL4J weight initialization enum
+     * @see WeightInit
+     */
+    public static WeightInit mapWeightInitialization(String kerasInit) {
+        /* WEIGHT INITIALIZATION
+         * TODO: finish mapping keras-to-dl4j weight distributions.
+         * Low priority since our focus is on loading trained models.
+         *
+         * Remaining dl4j distributions: DISTRIBUTION, SIZE, NORMALIZED,
+         * VI, RELU, XAVIER
+         */
+        WeightInit init = WeightInit.XAVIER;
+        if (kerasInit != null) {
+            switch (kerasInit) {
+                case "uniform":
+                    init = WeightInit.UNIFORM;
+                    break;
+                case "zero":
+                    init = WeightInit.ZERO;
+                    break;
+                case "lecun_uniform":
+                case "normal":
+                case "identity":
+                case "orthogonal":
+                case "glorot_normal":
+                case "glorot_uniform":
+                case "he_normal":
+                case "he_uniform":
+                default:
```

Comment:
Weight init, Keras -> DL4J mapping:
glorot_normal is xavier
glorot_uniform is xavier_uniform
he_normal is relu
he_uniform is relu_uniform
Normal and lecun uniform might map uniform and sigmoid uniform? check math on that